File: ai-diary/ai-diary/Models/HealthData.swift
import Foundation

struct HealthData: Codable {
    let id: String?
    let userId: String?
    let heartRate: Double?
    let sdnn: Double?
    let rmssd: Double?
    let pnn50: Double?
    let emotion: Int?
    let emotionLabel: String?
    let emotionConfidence: Double?
    let timestamp: Date
    let createdAt: Date?
}

File: backend/internal/models/health_data.go
package models

import (
	"github.com/google/uuid"
)

type HealthData struct {
	ID                uuid.UUID `json:"id"`
	UserID            uint      `json:"user_id"`
	HeartRate         *float64  `json:"heart_rate"`
	SDNN              *float64  `json:"sdnn"`
	RMSSD             *float64  `json:"rmssd"`
	PNN50             *float64  `json:"pnn50"`
	Emotion           *int      `json:"emotion"`
	EmotionLabel      *string   `json:"emotion_label"`
	EmotionConfidence *float64  `json:"emotion_confidence"`
	Timestamp         int64     `json:"timestamp"`
	CreatedAt         int64     `json:"created_at"`
}
